compiler/rustc_middle/src/ty/layout.rs

```diff
@@ -1671,27 +1671,25 @@ impl<'tcx> LayoutCx<'tcx, TyCtxt<'tcx>> {
         // always better to make the remaining locals ineligible, so we can
         // lay them out with the other locals in the prefix and eliminate
         // unnecessary padding bytes.
-        {
-            let mut used_variants = BitSet::new_empty(info.variant_fields.len());
-            for assignment in &assignments {
-                if let Assigned(idx) = assignment {
-                    used_variants.insert(*idx);
-                }
+        let mut used_variants = BitSet::new_empty(info.variant_fields.len());
```

Comment:
that doesn't necessarily seem desirable and shouldn't be linted iirc?

it causes us to drop `used_variants` later

Reply:
You're right this is a FP